File: django/frontend/src/components/views/character/pokemon/PokemonSkillsTab.js
import React, { Component } from "react";
import TabbedComponent from "./../../TabbedComponent";

class PokemonSkillsTab extends TabbedComponent {
	constructor(props) {
		super(props);
		this.tabRefs = {
			"self": React.createRef(),
		};
	}

	render() {
		const pokemon = this.props.pokemon;
		const species = pokemon.species;
		const classProp = this.props.class + " tabContent";
		return (
<div ref={this.tabRefs["self"]} className={classProp}>
	<table className="stat-block">
		<tbody>
			<tr>
				<th colSpan="2">Body</th>
			</tr>
			<tr>
				<td>Acrobatics</td>
				<td>{species.acrobatics}</td>
			</tr>
			<tr>
				<td>Athletics</td>
				<td>{species.athletics}</td>
			</tr>
			<tr>
				<td>Combat</td>
				<td>{species.combat}</td>
			</tr>
			<tr>
				<td>Intimidate</td>
				<td>{species.intimidate}</td>
			</tr>
			<tr>
				<td>Stealth</td>
				<td>{species.stealth}</td>
			</tr>
			<tr>
				<td>Survival</td>
				<td>{species.survival}</td>
			</tr>
		</tbody>
	</table>
	<table className="stat-block">
		<tbody>
			<tr>
				<th colSpan="2">Mind</th>
			</tr>
			<tr>
				<td>General Education</td>
				<td>{species.gen_education}</td>
			</tr>
			<tr>
				<td>Medical Education</td>
				<td>{species.med_education}</td>
			</tr>
			<tr>
				<td>Occult Education</td>
				<td>{species.occ_education}</td>
			</tr>
			<tr>
				<td>Pokemon Education</td>
				<td>{species.pok_education}</td>
			</tr>
			<tr>
				<td>Technology Education</td>
				<td>{species.tec_education}</td>
			</tr>
			<tr>
				<td>Guile</td>
				<td>{species.guile}</td>
			</tr>
			<tr>
				<td>Perception</td>
				<td>{species.perception}</td>
			</tr>
		</tbody>
	</table>
	<table className="stat-block">
		<tbody>
			<tr>
				<th colSpan="2">Spirit</th>
			</tr>
			<tr>
				<td>Charm</td>
				<td>{species.charm}</td>
			</tr>
			<tr>
				<td>Command</td>
				<td>{species.command}</td>
			</tr>
			<tr>
				<td>Focus</td>
				<td>{species.focus}</td>
			</tr>
			<tr>
				<td>Intuition</td>
				<td>{species.intuition}</td>
			</tr>
		</tbody>
	</table>
</div>
		);
	}
}

export default PokemonSkillsTab;

File: unity/Assets/Scripts/PokemonSkillsTab.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;

public class PokemonSkillsTab : MonoBehaviour
{
    private Pokemon pokemon;
    void Start()
    {

    }

    internal void Set(Pokemon pokemon)
    {
        this.pokemon = pokemon;
        Transform content = this.gameObject.transform;
        {
            Transform bodyPanel = content.Find("BodyPanel");
            bodyPanel.Find("Acr").gameObject.GetComponent<Text>().text = pokemon.species.acrobatics;
            bodyPanel.Find("Ath").gameObject.GetComponent<Text>().text = pokemon.species.athletics;
            bodyPanel.Find("Com").gameObject.GetComponent<Text>().text = pokemon.species.combat;
            bodyPanel.Find("Int").gameObject.GetComponent<Text>().text = pokemon.species.intimidate;
            bodyPanel.Find("Ste").gameObject.GetComponent<Text>().text = pokemon.species.stealth;
            bodyPanel.Find("Sur").gameObject.GetComponent<Text>().text = pokemon.species.survival;
        }
        {
            Transform mindPanel = content.Find("MindPanel");
            mindPanel.Find("Gen").gameObject.GetComponent<Text>().text = pokemon.species.gen_education;
            mindPanel.Find("Med").gameObject.GetComponent<Text>().text = pokemon.species.med_education;
            mindPanel.Find("Occ").gameObject.GetComponent<Text>().text = pokemon.species.occ_education;
            mindPanel.Find("Pok").gameObject.GetComponent<Text>().text = pokemon.species.pok_education;
            mindPanel.Find("Tec").gameObject.GetComponent<Text>().text = pokemon.species.tec_education;
            mindPanel.Find("Gui").gameObject.GetComponent<Text>().text = pokemon.species.guile;
            mindPanel.Find("Per").gameObject.GetComponent<Text>().text = pokemon.species.perception;
        }
        {
            Transform spiritPanel = content.Find("SpiritPanel");
            spiritPanel.Find("Cha").gameObject.GetComponent<Text>().text = pokemon.species.charm;
            spiritPanel.Find("Com").gameObject.GetComponent<Text>().text = pokemon.species.command;
            spiritPanel.Find("Foc").gameObject.GetComponent<Text>().text = pokemon.species.focus;
            spiritPanel.Find("Ins").gameObject.GetComponent<Text>().text = pokemon.species.intuition;
        }
    }
}
